scripts/globals/survival_guide.lua

```diff
@@ -0,0 +1,200 @@
+                favorites[10] = bit.rshift(option, 16) == 1 and 1 or 0
+            end
+
+            player:setTeleportMenu(travelType, favorites)
+        end
+
+        for x = 1, 3 do
+           favorites[1] = favorites[1] + favorites[x + 1] * 256 ^ x
+           favorites[5] = favorites[5] + favorites[x + 5] * 256 ^ x
+         end
+
+         favorites[9] = favorites[9] + favorites[10] * 256
+
+        player:updateEvent(favorites[1], favorites[5], favorites[9])
+    end
+end
+
+-------------------------------------------------
+-- public functions
+-------------------------------------------------
+
+dsp.survivalGuide.onTrigger = function(player)
+    local currentZoneId = player:getZoneID()
+    local tableIndex = zoneIdToMenuID[currentZoneId]
+    local guide = survivalGuides[tableIndex]
+
+    if guide then
+        -- Contains the zone to include, this is a bit-wise value.
+        local expansions = getEnabledExpansions()
+
+        -- If this survival guide hasn't been registered yet (saved to database) do that now.
+        local foundRegisteredGuide = checkForRegisteredSurvivalGuide(player,
+                                                                     guide)
+
+        if foundRegisteredGuide then
+            local param = bit.bor(tableIndex, bit.lshift(player:getCurrency("valor_point"), 16))
+
+            -- Get the teleport menu option.
+            -- Menu options can be organized by Region or Content.
+            -- Default (0) is region.
+            local teleportMenu = player:getTeleportMenu(travelType)
+            teleportMenu = teleportMenu[10]
+
+            if teleportMenu == 1 then
+                param = bit.bor(param, 0x0800)
+            end
+
+            if player:hasKeyItem(dsp.keyItem.RHAPSODY_IN_WHITE) == 1 then
+                -- "Rhapsody in White" key item reduces teleport fee by 80%
+                param = bit.bor(param, 0x2000)
+            end
+
+            local G1,G2,G3,G4 = unpack(player:getTeleport(travelType))
+
+            -- param 1 = Does nothing.
+            -- param 2 = current area, player amount of tabs, fee reducer(s) and menu layout (region/content).
+            -- param 3 = gil
+            -- param 4 = zones unlocked (group 1), set to -1 to enable all zones in the group.
+            -- param 5 = Zones unlocked (group 2), set to -1 to enable all zones in the group.
+            -- param 6 = Zones unlocked (group 3), set to -1 to enable all zones in the group.
+            -- param 7 = zones unlocked (Zehrun mines and Eastern Adoulin), set to -1 to enable all zones in the group.
+            -- param 8 = expansions available.
+            player:startEvent(cutsceneID, 0, param, player:getGil(), G1, G2, G3, G4, expansions)
+        end
+    else
+        player:PrintToPlayer('Survival guides are not enabled!')
+    end
+end
+
+dsp.survivalGuide.onEventUpdate = function(player, csid, option)
+    teleportMenuUpdate(player, option)
```

Comment:
Not sure if this is a question for me or for the group as a whole? Do you want me to change something here?

Reply:
Nah, so long as it functions as is, I _think_ you're fine.

Sometimes I have a habit of pointing things out just in case it becomes relevant in the future.

Reply:
Option 8 is used for homepoints, the survival guide equivalent is option 7. That said the function does respond to it with the favorites.